core/src/main/scala/org/broadinstitute/dsde/rawls/jobexec/WorkflowSubmissionActor.scala

```diff
@@ -412,6 +413,10 @@ trait WorkflowSubmission extends FutureSupport with LazyLogging with MethodWiths
       val executionServiceKey = workflowSubmitResult._1
       val executionServiceResults = workflowSubmitResult._2
 
+      // Emit metric for time until Cromwell has processed workflow.
+      val elapsedTime = System.currentTimeMillis() - submissionRec.submissionDate.getTime
+      workflowToCromwellLatency.update(elapsedTime, TimeUnit.MILLISECONDS)
```

Comment:
TOL: Anyone have thoughts about whether this should be in the `yield` or the last item in the `for` block? Or maybe even immediately after the `for ... yield`?

Reply:
Good questions. I don't think we want it _after_ the for loop because we are trying to capture the time for the individual workflows in batch. If I understand the yield correctly, it will be executed after each step in the for loop.

I think of the for comprehension as creating variables for subsequent steps, and this metric emitting is not relevant for future steps in the for comprehension. So I wouldn't be inclined to put it in the for unless my understanding of yield is incorrect, and the executing of the yield does not line up with the individual for loop executions.

Reply:
Also take everything I said with a huge grain of salt, given my limited Scala knowledge. If the yield should only be doing things directly related to the return values, moving the metric emitting into the for comprehension would be better. I'm really hoping y'all tell me what the right thing is to do here.

Reply:
FWIW, [here](https://github.com/broadinstitute/rawls/blob/f70467005d2e226ca62e2304cee9fa6619c4e7f5/core/src/main/scala/org/broadinstitute/dsde/rawls/jobexec/SubmissionSupervisor.scala#L167) is another example where it looks like yield is being used to collect statistics instead of returning something.

If putting the metric emission into the for comprehension is the better thing to do, I need some help with the syntax.

Reply:
> If I understand the yield correctly, it will be executed after each step in the for loop.

Actually it's the opposite. Each `->` in the `for` contents cause everything below it to iterate over every unpacked element. The yield is executed once and collects together all of the results.

So if you want one data point per batch, put it in the `yield`, but if you want one data point per workflow, put it at the end of the `for`.

If you want to put stuff at the end of the for it would be something like:

```
for {
  ...
  ... (existing content...)
  ...
  elapsedTime = System.currentTimeMillis() - submissionRec.submissionDate.getTime
  _ = workflowToCromwellLatency.update(elapsedTime, TimeUnit.MILLISECONDS)
}
```

Reply:
@cjllanwarne But the yield has to execute once for each call to `executionServiceCluster.submitWorkflows` since the yield depends on the result of that call (workflowSubmitResult). So I think it doesn't matter if the code is in the yield or in the for loop immediately after the call to `executionServiceCluster.submitWorkflows`. Am I missing something?

Reply:
Oh good point, I think you're right. TBH I don't normally put anything other than the result value in the `yield` so I probably got it wrong